### DISCUSSION TITLE
Convert remix.config.js into remix.config.ts

### DISCUSSION THREAD

**Discussion by MikeyUchiha** [remix-run/remix#2934] [Proposals]
For a project that's supposed to be Typescript focused, it's a little odd that when creating a base project it still uses a server.js file and remix.config.js file. I was able to convert my server.js to server.ts with some work but I can't seem to convert remix.config.js to remixconfig.json (or remixconfigrc like eslint and prettier) in order to make the project purely Typescript. Can you please provide a way to use remixconfig.json for making the project completely Typescript?

**Comment by kiliman** [comment]
You can add code to remix.config, not just configuration properties (like the routes config).

There's a rejected PR to support remix.config.ts https://github.com/remix-run/remix/pull/438

But the comments include a patch to add it if you want.

**Reply by MikeyUchiha** [reply]
@kiliman Would it be possible to JUST add support for it being a .ts file? I don't really feel comfortable using someone else's patch because it could potentially break from an update via Remix. It seems like a very small deal to allow remix.config.(js | ts).

**Reply by kiliman** [reply]
There's nothing wrong with making your OWN patch then. Open Source gives you the freedom to make your own changes. If you're dependent on somebody else, then you're going to have to settle for their priorities and not yours.

**Reply by MikeyUchiha** [reply]
Again, the point is that I would have to maintain it every single time there's an update which is ridiculous when all that it takes to change this is a few lines of code. Just seems really odd to support typescript only partially when it's a minor change to get it to work fully. I totally understand they may have other priorities but something that can be fixed in a few minutes, seems like a pretty easy win that would be helpful to quite a few people who have already asked about it.

**Comment by merelinguist** [comment]
It's not like JSON is any more type-safe. What's the big deal with having a JavaScript file?

**Reply by MikeyUchiha** [reply]
Same reason it's not a big deal to have it be a Typescript file. For me personally, I like my repository to look professional and consistent. I don't want to have some random Javascript file in my purely Typescript repository and I would change all my files from JS files to RC files as well. For someone who has little pride in their work and focuses on small projects then it's not a big deal, but for those of us who want their project to look professional and well maintained it is a big deal.

**Reply by kiliman** [reply]
> For someone who has little pride in their work and focuses on small projects then it's not a big deal, but for those of us who want their project to look professional and well maintained it is a big deal.

No offense, but that's a really bad take. Just because you're using Typescript doesn't make you professional just as not using it makes you any less.

Heck, the people that built Remix are highly professional and they're the ones that chose to use a JS file.

Anyway, questioning someone's professionalism is not a great way to get them to consider your request.

**Reply by MikeyUchiha** [reply]
I'm not saying people who use Typescript are more professional vs those who don't. I'm saying keeping code consistent looks more professional than code that's not consistent. For example, all of the files being Typescript vs some being Javascript and some not. Or having inconsistent naming conventions across files. Sorry I come from a consulting background where that is considered unprofessional no offense, but you're more than welcome to disagree.

**Reply by JClackett** [reply]
Dammm, wanting your project to be 100% TS so that 'looks' more professional is really not professional, that's OCD I'm afraid :(

**Reply by MikeyUchiha** [reply]
You're funny. So anyone who uses style guides, naming conventions, laws, regulations uniforms, structure, or any type of uniformity must all have OCD. Thanks for the laugh this morning. 😀

**Reply by JClackett** [reply]
I'm just saying that if you're reason for wanting TS is just so it "looks" good then I honestly wouldn't worry about it, it's just a file extension...(I also come from a "professional" consulting background). Saying that others who don't use it have "little pride in their work" really doesn't come across very well.

**Reply by MikeyUchiha** [reply]
It's for uniformity and professionalism. I feel sorry for your clients if that's how you do your work and refusing to follow the rules and structure that is put in place calling it "OCD". You're entitled to your opinion as I am mine. I take pride in my work being professional, uniform,  well written, modularized, extendable and consistent.

And that comment about just so it "looks" good proves you don't know much about consulting because looks, representation, and the user experience is what it's all about. But again, that's my opinion let's agree to disagree and stop derailing a very simple ask. Thanks.

**Reply by Loschcode** [reply]
Good news for this wonderful debate, I've a legit reason to want a TS rather than a JS in the config. I'm currently struggling to import some TS files that will be used in a custom routing that Remix allows (`defineRoutes`) and I can't make it work. Yes it's ridiculous to have a whole framework focused on TS and have some kind of base in JS, it's inconsistent and make things harder where you don't expect it.

**Reply by sergiodxa** [reply]
Now with Vite this file doesn’t exists and in Vite you can use TS.

**Comment by Ryongyon** [comment]
My feeling is that the official Remix team doesn't care about the feelings of the developers who use their framework. In addition to the "obsessive compulsive disorder" needs you said, I have also asked the official other questions and needs, such as: 

[https://github.com/remix-run/remix/discussions/4032](https://github.com/remix-run/remix/discussions/4032)
[https://github.com/remix-run/remix/issues/4257](https://github.com/remix-run/remix/issues/4257)
[https://github.com/remix-run/remix/discussions/4902](https://github.com/remix-run/remix/discussions/4902)

As you can see, even a lot of developers feel the same about issues and requirements, and until this year the Remix official team hasn't made any changes. 

I hope the official Remix team is aware of this:We love the Remix framework, and we ask questions or requests in the hope that it will get better and better, that it will become more comfortable to use, and that more developers will like and use Remix.
